src/test/java/org/jboss/logmanager/handlers/PeriodicRotatingFileHandlerTests.java

```
@@ -0,0 +1,146 @@
+        handler.publish(record);
+
+        Assert.assertTrue("File '" + logFile.getName() + "' does not exist", logFile.exists());
+
+        // Read the contents of the log file and ensure there's only one line and that like
+        List<String> lines = readAllLines(logFile);
+        Assert.assertEquals("More than 1 line found", 1, lines.size());
+        Assert.assertTrue("Expected the line to contain the date: " + currentDate, lines.get(0).contains(currentDate));
+
+        // Create a new record, increment the day by one and validate
+        cal.add(Calendar.DAY_OF_MONTH, nextDay);
+        final String nextDate = sdf.format(cal.getTime());
+        record = createLogRecord(Level.INFO, "Date: %s", nextDate);
+        record.setMillis(cal.getTimeInMillis());
+        handler.publish(record);
+
+        // Read the contents of the log file and ensure there's only one line and that like
+        lines = readAllLines(logFile);
+        Assert.assertEquals("More than 1 line found", 1, lines.size());
+        Assert.assertTrue("Expected the line to contain the date: " + nextDate, lines.get(0).contains(nextDate));
+
+        // The file should have been rotated as well
+        Assert.assertTrue("The rotated file '" + rotatedFile.getName() + "' does not exist", rotatedFile.exists());
+        lines = readAllLines(rotatedFile);
+        Assert.assertEquals("More than 1 line found", 1, lines.size());
+        Assert.assertTrue("Expected the line to contain the date: " + currentDate, lines.get(0).contains(currentDate));
+    }
+    
+    private List<String> readAllLines(File f) {
+        List<String> list =  new ArrayList();
+        
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(f));
+            String line;
+            while ((line = br.readLine()) != null) {
+               list.add(line);
+            }
+        }catch (Exception e) {
+            e.printStackTrace();
```

This should throw the error so the test will fail.